src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1480,6 +1482,27 @@ impl Coordinator {
         Ok(send_immediate_rows(vec![row]))
     }
 
+    pub(super) async fn sequence_inspect_shard(
+        &self,
+        session: &Session,
+        plan: InspectShardPlan,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        if !session.user().is_internal() {
+            return Err(AdapterError::Unauthorized(
+                rbac::UnauthorizedError::MzSystem {
+                    action: "inspect".into(),
+                },
+            ));
+        }
+        let state = self
+            .controller
+            .storage
+            .inspect_persist_state(plan.id)
+            .await?;
```

Comment:
Does this future have to be await'd in the same task? Can the future be moved to another task? We can probably leave this for now, but might be worth exploring that in the future to keep things off the coord thread when possible. I can do that in a followup.